drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -0,0 +1,497 @@
+/**
+ * Copyright (c) 2018 Linaro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define LOG_LEVEL CONFIG_SYS_LOG_WIFI_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wifi_eswifi_core);
+
+#define ESWIFI_DBG(fmt, ...) LOG_DBG("%s: " fmt, __func__, ##__VA_ARGS__)
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <errno.h>
+#include <gpio.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/net_context.h>
+#include <net/net_offload.h>
+#include <net/wifi_mgmt.h>
+
+#include <net/ethernet.h>
+#include <net_private.h>
+#include <net/net_core.h>
+#include <net/net_pkt.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <misc/printk.h>
+
+#include "eswifi.h"
+
+#define ESWIFI_WORKQUEUE_STACK_SIZE 1024
+K_THREAD_STACK_DEFINE(eswifi_work_q_stack, ESWIFI_WORKQUEUE_STACK_SIZE);
+
+static struct eswifi_dev eswifi0; /* static instance */
+
+static int eswifi_reset(struct eswifi_dev *eswifi)
+{
+	gpio_pin_write(eswifi->resetn.dev, eswifi->resetn.pin, 0);
+	k_sleep(10);
+	gpio_pin_write(eswifi->resetn.dev, eswifi->resetn.pin, 1);
+	gpio_pin_write(eswifi->wakeup.dev, eswifi->wakeup.pin, 1);
+	k_sleep(500);
+
+	/* fetch the cursor */
+	eswifi_request(eswifi, NULL, 0, eswifi->buf, sizeof(eswifi->buf));
+
+	return 0;
+}
+
+static inline int __parse_ssid(char *str, char *ssid)
+{
+	/* fnt => '"SSID"' */
+
+	if (!*str || (*str != '"')) {
+		return -EINVAL;
+	}
+
+	str++;
+	while (*str && (*str != '"')) {
+		*ssid++ = *str++;
+	}
+
+	*ssid = '\0';
+
+	if (*str != '"') {
+		return -EINVAL;
+	}
+
+	return -EINVAL;
+}
+
+static void __parse_scan_res(char *str, struct wifi_scan_result *res)
+{
+	int field = 0;
+
+	/* fmt => #001,"SSID",MACADDR,RSSI,BITRATE,MODE,SECURITY,BAND,CHANNEL */
+
+	while (*str) {
+		if (*str != ',') {
+			str++;
+			continue;
+		}
+
+		if (!*++str) {
+			break;
+		}
+
+		switch (++field) {
+		case 1: /* SSID */
+			__parse_ssid(str, res->ssid);
+			res->ssid_length = strlen(res->ssid);
+			str += res->ssid_length;
+			break;
+		case 2: /* mac addr */
+			break;
+		case 3: /* RSSI */
+			res->rssi = atoi(str);
+			break;
+		case 4: /* bitrate */
+			break;
+		case 5: /* mode */
+			break;
+		case 6: /* security */
+			if (!strncmp(str, "Open", 4)) {
+				res->security = WIFI_SECURITY_TYPE_NONE;
+			} else {
+				res->security = WIFI_SECURITY_TYPE_PSK;
+			}
+			break;
+		case 7: /* band */
+			break;
+		case 8: /* channel */
+			res->channel = atoi(str);
+			break;
+		}
+
+	}
+}
+
+struct eswifi_dev *eswifi_by_iface_idx(u8_t iface)
+{
+	/* only one instance */
+	ESWIFI_DBG("%d", iface);
+	return &eswifi0;
+}
+
+static int __parse_ipv4_address(char *str, char *ssid, u8_t ip[4])
```

Comment:
Note that there exists also a function to parse IPv4 address.
`bool net_ipaddr_parse(const char *str, size_t str_len, struct sockaddr *addr)`
Not sure if it would suit your use case here.

Reply:
thanks, that could help indeed, anyway I plan to go over all parsing functions to make them more factorized after this basic support is merged.